### PR TITLE
fix(sso): support for SSL update && ensure update is atomic

### PR DESCRIPTION
**Pull request by lafirest**
1. support update SSL key and cert files
2. increase connection timeout
3. ensure the update is atomicity, everything will be consistent

Fixes [EMQX-11037](https://emqx.atlassian.net/browse/EMQX-11037) [EMQX-11045](https://emqx.atlassian.net/browse/EMQX-11045)

### <samp>🤖 Generated by Copilot at b2699c6</samp>

This pull request improves the SSO backend management and testing for LDAP. It increases the timeout, protects sensitive data, handles SSL files, and ensures atomic updates. It also refactors some code and fixes a default value inconsistency.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl [hunk 1]
```diff
@@ -175,47 +184,60 @@ start_backend_services() ->
                     ?SLOG(error, #{
                         msg => "start_sso_backend_failed",
                         backend => Backend,
-                        reason => Reason
+                        reason => emqx_utils:redact(Reason)
```
Comment (zmstone) [inline apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl:187]: insert ets with a empty map `#{}` as state, and `last_error => emqx_utils:redact(Reason)` ?

File: apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl [hunk 2]
```diff
@@ -175,47 +184,60 @@ start_backend_services() ->
...
+            case Result of
+                ok ->
+                    OkFun(Result);
+                {ok, _} ->
+                    OkFun(Result);
+                {error, _} = Error ->
+                    ErrFun(Error)
+            end;
+        {error, _} = Error ->
+            ErrFun(Error)
```
Comment (zmstone) [inline apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl:229]:
```suggestion
                {error, Reason} ->
                    {error, ErrFun(Error)}
            end;
        {error, Reason} ->
            {error, ErrFun(Error)}
```

File: apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl [hunk 3]
```diff
@@ -175,47 +184,60 @@ start_backend_services() ->
...
-    {ok, Config};
+    maybe_write_certs(Config);
 pre_config_update(_, {delete, _Backend}, undefined) ->
     throw(not_exists);
 pre_config_update(_, {delete, _Backend}, _OldConf) ->
     {ok, null}.
 
 post_config_update(_, UpdateReq, NewConf, _OldConf, _AppEnvs) ->
-    call({update_config, UpdateReq, NewConf}).
+    {ok, on_config_update(UpdateReq, NewConf)}.
```
Comment (zmstone) [inline apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl:240]: if we are to ignore errors:
```suggestion
    _ = on_config_update(UpdateReq, NewConf),
    ok.
```

File: apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl [hunk 4]
```diff
@@ -175,47 +184,60 @@ start_backend_services() ->
...
-            {error,
-                case Reason of
-                    {_Stage, _Mod, Reason2} ->
-                        Reason2;
-                    _ ->
-                        Reason
-                end}
+            case Result of
+                ok ->
+                    OkFun(Result);
```
Comment (zmstone) [inline apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl:222]:
```
log_ok()
case get_backend_status(Backend) of
    ok -> ok;
    {error, Reason} ->
        %% a negative feedback for the caller
        {error, #{msg => "config_updated_but_failed_to_start_backend", reason => Reason}}
end.
``

File: apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl [hunk 5]
```diff
@@ -331,3 +351,66 @@ add_handler() ->
...
+new_ssl_source(Source, undefined) ->
+    Source;
+new_ssl_source(Source, SSL) ->
+    Source#{<<"ssl">> => SSL}.
+
+clear_start_error() ->
+    mark_start_error(<<>>).
+
+mark_start_error(Reason) ->
+    erlang:put(?START_ERROR_KEY, Reason).
```
Comment (zmstone) [inline apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl:380]: can we insert ets directly here?

File: apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl [hunk 6]
```diff
@@ -170,52 +179,53 @@ start_backend_services() ->
...
-                msg => "update_sso_successfully",
-                backend => Backend,
-                result => Result
-            }),
-            Result;
-        {error, Reason} ->
+        {ok, _UpdateResult} ->
+            case lookup(Backend) of
+                undefined ->
+                    ok;
```
Comment (zmstone) [inline apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl:203]: Q: how can this happen?

**Review by zmstone — APPROVED** [review]

File: apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl (hunk 6 above)
Comment (lafirest) [reply apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl:203]: when deleting the backend

**Review by HJianBo — APPROVED** [review]

**Pull request merged**